spec/enumerable_spec.rb

```diff
@@ -0,0 +1,202 @@
+require_relative('../enumerable')
+
+describe ::Enumerable do
+  describe "#my_each" do
+    it "Loops An Enumerable" do
+      array = []
+      %w[a b c].my_each { |item| array << item }
+      expect(array).to(eq(%w[a b c]))
+    end
+
+    it "Loops An Range" do
+      array = []
+      (1...4).my_each { |item| array << item }
+      expect(array).to(eq([1, 2, 3]))
+    end
+
+    it "Loops An Hash" do
+      array = []
+      { "Jane Doe" => 10, "Jim Doe" => 6 }.my_each { |item| array << item[0] }
+      expect(array).to(eq(["Jane Doe", "Jim Doe"]))
+    end
+  end
```

Comment:
Please add a negative scenario for this spec which can be something like:  `it 'should not return an element that is not in the array'`